File: module_test_1/src/main/org/grynko/nazar/java/Main.java
package main.org.grynko.nazar.java;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Main {

    public static void main(String[] args) {
        List<BusStop> busStops = new ArrayList<>();
        Random random = new Random();
        int stopNumber = 5;
        int busNumber = 7;
        int busStopBorder = 5;

        for(int i = 0; i < stopNumber; ++i) {
            int randomBorder = random.nextInt(busStopBorder) + 1;
            String name = String.format("Stop-%d", i + 1);
            BusStop busStop = new BusStop(randomBorder, name);

            busStops.add(busStop);
        }

        for(int i = 0; i < busNumber; ++i) {
            Bus bus = new Bus(busStops);
            String name = String.format("Bus-%d", i + 1);

            new Thread(bus, name).start();
        }
    }

}

File: module_test_1/src/main/org/grynko/nazar/go/main.go
package main

import (
	"Lab_3/module_test/bus"
	"Lab_3/module_test/bus_stop"
	"fmt"
	"math/rand"
	"time"
)

func main() {
	stopNumber := 5
	busNumber := 7
	busStopBorder := 5

	stops := make([]bus_stop.BusStop, stopNumber)

	for i := 0; i < stopNumber; i++ {
		randomBorder := rand.Intn(busStopBorder) + 1
		name := fmt.Sprintf("Stop-%d", i+1)
		busStop := bus_stop.Init(randomBorder, name)

		stops[i] = *busStop
	}

	for i := 0; i < busNumber; i++ {
		name := fmt.Sprintf("Bus-%d", i+1)
		busImp := bus.Init(stops, name)

		go busImp.Run()
	}
	
	time.Sleep(60 * time.Minute)
}
